express-notes-api/index.js

```diff
@@ -0,0 +1,150 @@
+const express = require('express');
+const app = express();
+const data = require('./data.json');
+const JSONMiddleware = express.json();
+app.use(JSONMiddleware);
+
+// require automatically parses
+
+app.get('/api/notes', function (req, res) {
+  const array = [];
+  for (const property in data.notes) {
+    array.push(data.notes[property]);
+  }
+  res.status(200);
+  res.send(array);
+});
+
+app.get('/api/notes/:id', function (req, res) {
+  const reqId = Number(req.params.id);
+  if (isNaN(reqId)) {
+    const errorObject = {
+      error: 'id must be a positive integer'
+    };
+    res.send(errorObject);
+    res.sendStatus(400);
+    return;
+  }
+  const array = [];
+  for (const property in data.notes) {
+    if (data.notes[property].id === reqId) {
+
+      array.push(data.notes[property]);
+
+    } else {
+      const cannotFindObject = {
+        error: `cannot find note with id ${reqId}`
+      };
+      res.send(cannotFindObject);
+      res.sendStatus(404);
+
+      return;
+    }
+
+    res.send(array[0]);
+
+  }
+
+});
+let nextId = data.nextId;
+
+app.post('/api/notes', function (req, res) {
+  const reqBody = req.body;
+  if (!reqBody.content) {
+    const errorObject = {
+      error: 'content is a required field'
+    };
+    res.status(400);
+    res.send(errorObject);
+  } else if (reqBody.content) {
+    reqBody.id = nextId;
+    data.notes[nextId] = reqBody;
+    nextId++;
+    res.status(201);
+    res.send(reqBody);
```

Comment:
In here, you'll want to use `fs.writeFile` to make saving the new object permanent.